test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1691,173 +1695,101 @@ def test_multiple_equivalence_class_preemption(self):
                                  max_attempts=10, starttime=self.t)
 
-        # delete the expressq jobs and test that the suspended jobs
-        # go back to running state. equivalence classes=4 again
-        self.server.deljob(jid5, wait=True)
-        self.server.deljob(jid6, wait=True)
-
-        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
-        self.server.expect(JOB, {'job_state': 'S'}, id=jid2)
-        self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
-        self.server.expect(JOB, {'job_state': 'R'}, id=jid4)
+        # Resume the jobs suspended by qsig
+        # 1 second delay is added so that time of next logging moves ahead.
+        # This will make sure log_match does not take previous entry.
+        time.sleep(1)
+        self.server.sigjob(jobid=jid1, signal="resume")
+        self.server.sigjob(jobid=jid3, signal="resume")
 
-        # Check equivalence classes =4
-        self.scheduler.log_match("Number of job equivalence classes: 4",
+        # On resume check that there are same number of equivalence classes
+        t = int(time.time())
+        self.scheduler.log_match("Number of job equivalence classes: 3",
                                  max_attempts=10, starttime=self.t)
 
-    def test_held_jobs_equiv_class(self):
-        """
-        Test that held jobs do not go into another equivalence class.
-        """
-
-        a = {'resources_available.ncpus': 2}
-        self.server.create_vnodes('vnode', a, 1, self.mom, usenatvnode=True)
-
-        a = {'Resource_List.ncpus': 1}
-
-        (jid1, jid2, jid3) = self.submit_jobs(3)
+        # delete the expressq jobs and check that the suspended jobs
+        # go back to running state. equivalence classes=2 again
+        self.server.deljob(jid5, wait=True)
 
         self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
-        self.server.expect(JOB, {'job_state': 'Q'}, id=jid3)
-
-        self.server.holdjob(jid1)
-        self.server.rerunjob(jobid=jid1, runas=ROOT_USER)
-        self.server.expect(JOB, {'job_state': 'H'}, id=jid1)
-
         self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid4)
 
-        self.scheduler.log_match("Number of job equivalence classes: 1",
+        # Check equivalence classes =2
+        t = int(time.time())
+        self.scheduler.log_match("Number of job equivalence classes: 2",
                                  max_attempts=10, starttime=self.t)
 
-    def test_multiple_node_multiple_job_order(self):
+    def test_held_jobs_equiv_class(self):
         """
-        Test jobs that come later in query order on multi-node
-        setup are added to its own equivalence class.This is
-        multinode test for test_premption2
+        1) Test that held jobs do not go into another equivalence class.
+        2) Test that queued jobs do not go to different equivalence class
```

Comment:
Queued jobs by default don't go into another equivalence class.  Change this to running jobs do not go into another equivalence class.